### PR TITLE
Jarv/no deploy

### PR DESCRIPTION
**Pull request by jarv**
This combines the other PR to remove the deploy tags, replacing them a `- include: deploy.yml tags=deploy`. This also makes some additional tweaks so we can now run `--skip-tags deploy` on a fresh ubuntu 12.04 server which will mean that all the `main.yml` tasks will install without running any of the `deploy.yml` tasks. To do this there is a `<role>_installed` fact set at the end of deploy.yml that needs to be defined for notification handlers.

### CODE REVIEW

File: util/jenkins/ansible-provision.sh [hunk 1]
```diff
@@ -75,7 +75,7 @@ if [[ -z $instance_type ]]; then
   if [[ $server_type == "full_edx_installation" ]]; then
     instance_type="m1.medium"
   elif [[ $server_type == "ubuntu_12.04" ]]; then
-    instance_type="m1.small"
```
Comment (e0d) [inline util/jenkins/ansible-provision.sh:78]: Why not a parameter?

File: playbooks/roles/edxapp/tasks/deploy.yml [hunk 2]
```diff
@@ -325,4 +295,5 @@
   when: celery_worker is defined and not devstack
   with_items: edxapp_workers
   sudo_user: "{{ supervisor_service_user }}"
-  tags: deploy
+
+- set_fact: edxapp_installed=true
```
Comment (e0d) [inline playbooks/roles/edxapp/tasks/deploy.yml:299]: Seems like three service do this now.  Should this be universalized?

File: util/jenkins/ansible-provision.sh (hunk 1 above)
Comment (jarv) [inline util/jenkins/ansible-provision.sh:78]: I'll remove this logic since I want to default to m1.medium for both server types.

Comment (carsongee) [inline util/jenkins/ansible-provision.sh:78]: Don't you want m3.medium now ;)

File: playbooks/roles/edxapp/tasks/deploy.yml (hunk 2 above)
Comment (jarv) [inline playbooks/roles/edxapp/tasks/deploy.yml:299]: universalized how?

**Pull request merged**